robotiq_s_model_control/src/s_model_ethercat_node.cpp

```diff
@@ -0,0 +1,162 @@
+    }
+
+    /** \brief Timer event
+     *         Note: we do not use the TimerEvent time difference because it does NOT guarantee that
+     * the time source is
+     *         strictly linearly increasing
+     */
+    void update(const ros::TimerEvent& e)
+    {
+        //! Get change in time
+        clock_gettime(CLOCK_MONOTONIC, &current_time_);
+        elapsed_time_ =
+                ros::Duration(current_time_.tv_sec - last_time_.tv_sec + (current_time_.tv_nsec - last_time_.tv_nsec) / BILLION);
+        last_time_ = current_time_;
+        ROS_DEBUG_STREAM_THROTTLE_NAMED(1, "GenericHWLoop","Sampled update loop with elapsed time " << elapsed_time_.toSec());
+
+        //! Error check cycle time
+        const double cycle_time_error = (elapsed_time_ - desired_update_freq_).toSec();
+        if (cycle_time_error > cycle_time_error_threshold_)
+        {
+            ROS_WARN_STREAM_NAMED(name_, "Cycle time exceeded error threshold by: "
+                                  << cycle_time_error << ", cycle time: " << elapsed_time_
+                                  << ", threshold: " << cycle_time_error_threshold_);
+        }
+
+        //! Input
+        hardware_interface_->read(elapsed_time_);
+
+        //! Control
+        controller_manager_->update(ros::Time::now(), elapsed_time_);
+
+        //! Output
+        hardware_interface_->write(elapsed_time_);
+    }
+
+protected:
+    // Startup and shutdown of the internal node inside a roscpp program
+    ros::NodeHandle nh_;
+
+    // Name of this class
+    std::string name_;
+
+    // Settings
+    ros::Duration desired_update_freq_;
+    double cycle_time_error_threshold_;
+
+    // Timing
+    ros::Timer non_realtime_loop_;
+    ros::Duration elapsed_time_;
+    double loop_hz_;
+    struct timespec last_time_;
+    struct timespec current_time_;
+
+    /** \brief ROS Controller Manager and Runner
+     *
+     * This class advertises a ROS interface for loading, unloading, starting, and
+     * stopping ros_control-based controllers. It also serializes execution of all
+     * running controllers in \ref update.
+     */
+    boost::shared_ptr<controller_manager::ControllerManager> controller_manager_;
+
+    /** \brief Abstract Hardware Interface for your robot */
+    boost::shared_ptr<robotiq_s_model_control::SModelHWInterface> hardware_interface_;
+};
+
+int main(int argc, char** argv)
+{
+
+    using robotiq_ethercat::EtherCatManager;
+    using robotiq_s_model_control::SModelEtherCatClient;
+    ros::init(argc, argv, "robotiq_s_model_gripper_node");
+    ros::NodeHandle nh;
+    ros::NodeHandle pnh("~");
+
+    // NOTE: We run the ROS loop in a separate thread as external calls such
+    // as service callbacks to load controllers can block the (main) control loop
+    ros::AsyncSpinner spinner(2);
+    spinner.start();
+
+    // Parameter names
+    std::string ifname;
+    int slave_no;
+    bool activate;
+
+    nh.param<std::string>("ifname", ifname, "enp9s0");
```

Comment:
In the past, I know this default has been something like `eth1`. Can we return it to something along those lines?